ynput/ayon-deadline
# Add USD pinning environments to submitters

## Changelog Description
This PR is adding USD pinning related environments to specific submitters so they can pass it to the Deadline jobs. Mainly it is setting `ENABLE_STATIC_GLOBAL_CACHE` and adding **rootless** path to pinning json file as `PINNING_FILE_PATH`. All of it only if the pinning file is actually created and published. Note, that this pinning file path - since it is rootless - cannot be directly used by the resolver. Therefor when job is started, pre-launch hook from usd addon resolves the rootless path to absolute path and set `PROJECT_ROOTS` environment variable. With that, resolver should be able to use the pinning file.

## Testing notes:

Use it in combination with https://github.com/ynput/ayon-usd/pull/41
- set your usd scene
- submit it to farm

> [!note]
> I consider these changes temporary. We need better handling of job environments. See
> https://github.com/ynput/ayon-core/issues/876
